js/packages/teams-ai/src/TestMemoryFork.ts

```diff
@@ -0,0 +1,82 @@
+import { Memory } from './MemoryFork';
+
+/**
+ * @private
+ */
+const TEMP_SCOPE = 'temp';
+
+/**
+ * A test version of the Memory class used by unit tests.
+ */
+export class TestMemoryFork implements Memory {
+    private readonly _fork: Record<string, Record<string, unknown>> = {};
```

Comment:
why do we need a `TestMemoryFork`?

Reply:
`MemoryFork` acts as a forked memory (e.g. copy) of the original memory, and so its [constructor](https://github.com/microsoft/teams-ai/pull/1051/files/e85ff466d0e61dd68c9cce523713eaf89e231a05#diff-5a43a10e2ff885cd98465b5aa99266365ce0e36be5b99b1550484555aa4f85bcL65) takes in an instance of itself. Adding `TestMemoryFork` acts as the original memory, and allows us to simulate the tests with both

Reply:
since the seem identical couldn't we just use two instances of `MemoryFork` instead of having another class just for tests that is essentially a copy and paste (unless im missing some hidden functionality). Or `TestMemoryFork` could `extend` `MemoryFork` so it could be centralized.

Reply:
it is minimal but I tweaked some of the logic in `hasValue` and `getValue` so that instead of checking the original memory, they return false/null instead

Reply:
I would recommend changing the tests and just using the `MemoryFork` to test itself, from my perspective the `TestMemoryFork` seems unnecessary since a memory fork can accept an instance of itself as a parent constructor argument.  The less code we have the easier it is to maintain.

Reply:
resolved offline- unable to remove `TestMemoryFork` as constructor requires argument and memory errors will be thrown otherwise